### ISSUE TITLE
Difference between TransformerEncoderLayer and TransformerSentenceEncoderLayer

### ISSUE THREAD

**Issue by lematt1991** [pytorch/fairseq#777]
These two modules seem to be equivalent (as long as `normalize_before=False` in `TransformerEncoderLayer`), or am I missing something.  Do we need both of them?  CC @kartikayk @myleott 

[TransformerSentenceEncoderLayer](https://github.com/pytorch/fairseq/blob/master/fairseq/modules/transformer_sentence_encoder_layer.py)
[TransformerEncoderLayer](https://github.com/pytorch/fairseq/blob/master/fairseq/models/transformer.py#L495)

**Comment by myleott** [comment]
We don't need both -- I have a TODO somewhere to merge them, but love if you have cycles to do this :) The main thing is we may need to define an upgrade_state_dict method so that old checkpoints continue to work
